src/parse.rs

```diff
@@ -150,76 +192,74 @@ impl_fromhex_array!(512);
 
 /// Hex decoding error.
 #[derive(Debug, Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
-pub enum Error {
-    /// Non-hexadecimal character.
-    InvalidChar(u8),
-    /// Purported hex string had odd length.
-    OddLengthString(usize),
-    /// Tried to parse fixed-length hash from a string with the wrong type (expected, got).
+pub enum HexToArrayError {
+    /// Conversion error while parsing hex string.
+    Conversion(HexToBytesError),
+    /// Tried to parse fixed-length hash from a string with the wrong length (got, want).
     InvalidLength(usize, usize),
```

Comment:
In 118ee3acb1e3c431497f6c9a753458d38d2d1a7c

I have a mild preference for making `got` and `want` named fields. But this is fine.

Reply:
I've done that somewhere in the last week, I remember because I used `expected` and `got`, and I wanted to go back and change it to `expected` and `actual` (I favour got and want in unit tests but thought the longer ones were better for field names, do you have a preference?)